File: content/examples/other/leetcode/problems/longest-consecutive-sequence/example-2/solution.js
function longestConsecutiveHashSet(nums) {
  const set = new Set(nums);
  let best = 0;
  for (const x of set) {
    if (set.has(x - 1)) continue;
    let cur = 1;
    while (set.has(x + cur)) cur += 1;
    best = Math.max(best, cur);
  }
  return best;
}

if (require.main === module) {
  console.log(longestConsecutiveHashSet([100, 4, 200, 1, 3, 2]));
}

module.exports = { longestConsecutiveHashSet };

File: content/examples/other/leetcode/problems/longest-consecutive-sequence/example-2/solution.py
def longest_consecutive_hashset(nums):
    s = set(nums)
    best = 0
    for x in s:
        if x - 1 in s:
            continue
        cur = 1
        while x + cur in s:
            cur += 1
        best = max(best, cur)
    return best


if __name__ == "__main__":
    print(longest_consecutive_hashset([100, 4, 200, 1, 3, 2]))
